Chime/Parser.m

```diff
@@ -233,11 +322,22 @@ - (NSMutableArray *)passTwo {
                     fill:bytecodes];
       }
     }
-    _counter++;
+    _physicalLinesCounter++;
   }
   return bytecodes;
 }
 
+NSArray *cleanProgram(NSString *stringProgram) {
+  NSMutableArray *trimmedProgram = [NSMutableArray array];
+  for (NSString *line in [stringProgram componentsSeparatedByString:@"\n"]) {
+    NSString *trimmedLine =
+        [line stringByTrimmingCharactersInSet:[NSCharacterSet
+                                                  whitespaceCharacterSet]];
+    [trimmedProgram addObject:trimmedLine];
+  }
+  return [trimmedProgram copy];
```

Comment:
Is there any particular reason we are copying the array instead of just returning it?

Reply:
I believe this has something to do with converting `NSMutableArray` to `NSArray`.